contrib/resources/mapping/MAIN.TXT

```diff
@@ -0,0 +1,2428 @@
+0xfd 0x20ac        #EURO SIGN
+EXTENDS CODEPAGE 1125
+
+CODEPAGE 849 # Belarusian, Cyrillic, with EUR
+# reference: https://en.wikipedia.org/wiki/Code_page_866
+0xfb 0x20ac        #EURO SIGN
+EXTENDS CODEPAGE 1131
+
+CODEPAGE 850 # Latin-1 (Western European)
+EXTENDS FILE mapping-unicode.org CP_850.TXT
+
+CODEPAGE 851 # Greek, old encoding
+EXTENDS FILE mapping-wikipedia.org CP_851.TXT
+
+CODEPAGE 852 # Latin-2 (Central European), with EUR
+0xaa 0x20ac        #EURO SIGN - FreeDOS specific
+EXTENDS FILE mapping-unicode.org CP_852.TXT
+
+CODEPAGE 853 # Latin-3 (Turkish, Maltese, and Esperanto)
+0xaa 0x20ac        #EURO SIGN       - FreeDOS specific
+0xd0 0x00a2        #CENT SIGN       - FreeDOS specific
+0xd1 0x00a5        #YEN SIGN        - FreeDOS specific
+0xf1 0x00b1        #PLUS-MINUS SIGN - FreeDOS specific
+0xdd 0x20a4        #LIRA SIGN       - FreeDOS specific
+0xfb 0x20b7        #SPESMILO SIGN   - FreeDOS specific
+EXTENDS FILE mapping-wikipedia.org CP_853.TXT
+
+CODEPAGE 855 # South Slavic, Cyrillic
+EXTENDS FILE mapping-unicode.org CP_855.TXT
+
+CODEPAGE 856 # Hebrew-2, with EUR
+0xad 0x20ac        #EURO SIGN - FreeDOS specific
+EXTENDS FILE mapping-unicode.org CP_856.TXT
+
+CODEPAGE 857 # Latin-5, with EUR
+0xd5 0x20ac        #EURO SIGN - FreeDOS specific
+EXTENDS FILE mapping-unicode.org CP_857.TXT
+
+CODEPAGE 858 # Latin-1 (Western European), with EUR
+0xd5 0x20ac        #EURO SIGN - FreeDOS specific
+EXTENDS CODEPAGE 850
+
+CODEPAGE 859 # Latin-9 (Western European), with EUR
+# reference: https://en.wikipedia.org/wiki/Code_page_859
+0xab 0x0153        #LATIN SMALL LIGATURE OE
+0xac 0x0152        #LATIN CAPITAL LIGATURE OE
+0xd5 0x20ac        #EURO SIGN
+0xdd 0x0160        #LATIN CAPITAL LETTER S WITH CARON
+0xef 0x017d        #LATIN CAPITAL LETTER Z WITH CARON
+0xf3 0x0178        #LATIN CAPITAL LETTER Y WITH DIAERESIS
+0xf7 0x017e        #LATIN SMALL LETTER Z WITH CARON
+0xf9 0x0161        #LATIN SMALL LETTER S WITH CARON
+EXTENDS CODEPAGE 850
+
+CODEPAGE 860 # Portuguese
+EXTENDS FILE mapping-unicode.org CP_860.TXT
+
+CODEPAGE 861 # Icelandic
+EXTENDS FILE mapping-unicode.org CP_861.TXT
+
+CODEPAGE 862 # Hebrew-2
+EXTENDS FILE mapping-unicode.org CP_862.TXT
+
+CODEPAGE 863 # Canadian French
+EXTENDS FILE mapping-unicode.org CP_863.TXT
+
+CODEPAGE 864 # Arabic
+# 0x9b TODO: unidentified character - FreeDOS specific
+# 0x9c TODO: unidentified character - FreeDOS specific
+# 0x9f TODO: unidentified character - FreeDOS specific
+# 0xa6 TODO: unidentified character - FreeDOS specific
+# 0xa7 TODO: unidentified character - FreeDOS specific
+EXTENDS FILE mapping-unicode.org CP_864.TXT
+
+CODEPAGE 865 # Nordic
+EXTENDS FILE mapping-unicode.org CP_865.TXT
+
+CODEPAGE 866 # Russian, Cyrillic
+# Note: this is NOT an alternative layout mentioned here:
+# https://en.wikipedia.org/wiki/Code_page_866
+EXTENDS FILE mapping-unicode.org CP_866.TXT
+
+CODEPAGE 867 # Czech and Slovak, Kamenicky encoding
```

Comment:
FreeDOS uses both 867 and 895 as Kamenický encoding. As stated above - both 867 and 895 are sometimes used for other code pages. Since our CPI/CPX (screen font) files and keyboard layouts come exclusively from FreeDOS, we have to respect it's numbering scheme.